components/capi/src/macros.rs

```diff
@@ -0,0 +1,49 @@
+macro_rules! c_enum {
```

Comment:
I added one of these for enums but not structs because structs have the additional complexity that each _field_ can need `Into`, which we can solve by either:

 - call `into()` on everything (allows for changes in the field type to get missed)
 - allow an `is` syntax in the field level, which does make the macro trickier to write (but still doable)